### ISSUE TITLE
How to install ?

### ISSUE THREAD

**Issue by thunderpants** [Elinvention/efiboots#38]
Noob question.

It doesn't seem clear to me how this is meant to be installed.
I can see instructions for adding dependencies but not for how to install the main gui program.

**Comment by Elinvention** [comment]
Thanks for your question. You are right, the README assumes that one is familiar with running python applications. You don't need a deb package or nix.
You can clone this repo if you know how to do that with git or [download the source code](https://github.com/Elinvention/efiboots/archive/refs/heads/main.zip) and unzip it. Then double click efiboots (make it executable before if it isn't already).

**Comment by KaKi87** [comment]
Hi,
Would you consider making builds ?
Thanks

**Comment by Elinvention** [comment]
What do you mean by builds? Deb packages?

**Comment by KaKi87** [comment]
DEB, PPA, AppImage, or a binary of any form, would be nice, yes please.

**Comment by codefaux** [comment]
Here to ask for the same. AppImage would be best, runs on Any Linux, just download and chmod +x.

**Comment by KaKi87** [comment]
> AppImage would be best, runs on Any Linux, just download and chmod +x

Flatpak would be best then, because you get automated updates in addition to the ability to run on any Linux.

Thanks

**Comment by codefaux** [comment]
> > AppImage would be best, runs on Any Linux, just download and chmod +x
> 
> Flatpak would be best then, because you get automated updates in addition to the ability to run on any Linux.
> 
> Thanks

This expresses a distinct misunderstanding with how AppImage works.

AppImage = download one file, chmod +x, execute, running app.

flatpak = install flatpak management/container/downloader service, download application container + all application container dependencies, assemble and launch, with permissions, restrictions, inaccessible folders and devices, manual mappings, etc etc etc

Appimage = ANY LINUX
Flatpak = One of 41 selected Linuxen, after installation of Flatpak management services

AppImage would be best, runs on __Any Linux__, just downlaod and chmod +x.

**Comment by KaKi87** [comment]
> This expresses a distinct misunderstanding with how AppImage works.
> 
> AppImage = download one file, chmod +x, execute, running app.

Lol, tell me something I don't know.

> flatpak = install flatpak management/container/downloader service, download application container + all application container dependencies, assemble and launch, with permissions, restrictions, inaccessible folders and devices, manual mappings, etc etc etc

That's plain bad faith.

I'm trying to tell you : Flatpak provides automated updates.

As in : you don't need to re-download an AppImage every time there's an update.

> 41 Linuxen claim; https://flatpak.org/ and https://flatpak.org/setup/ -- specifically not Any Linux

If your distro isn't among those, then you're not a normal user anyway, so you can compile a program from sources.

This issue is for normal users, who use easy distros. Distros that are supported by Flatpak.

**Comment by codefaux** [comment]
> tell me something I don't know. (And some other bullshit stripped for brevity)

Bro your argument was any Linux and it was patently false in every way. Don't be a child. I'm not arguing with children on GitHub. Let's keep on topic here. Yeah flatpak and its massive orchestration service architecture handles updates.  Congrats. That's not what I said, and it's separate from what you claimed. 

Yes, you have to update appimage yourself. (Unless you use an appimage launcher which handles autoupdates too, without an orchesteation service.)

Flatpak runs on fewer systems, with more overhead for the user, the user's hardware, the developer, and the internet in general.  It comes from a centralized repo, and depends on all of the behind the scenes flatpak services and subsystems. It limits the access and reach of the binaries within it by default.

AppImage is one compiled binary provided by the developer from GitHub Releases or literally ant other file transfer medium, downloaded and executed by itself. It's an extra makefile step, not signing up with a flatpack repo and keeping that up to date.

There's no choice if you're looking for reaching Any User on Any Linux.

**Comment by KaKi87** [comment]
> There's no choice if you're looking for reaching Any User on Any Linux.

But we're not.

We want builds and automated updates because we're normal users using easy distros that Flatpak does support.

We're not using an exotic distro that targets advanced users who wouldn't need builds because having that exotic distro implies they're comfortable with building apps themselves.

**Comment by codefaux** [comment]
> But we're not.

Oh, fair enough. I got the impression otherwise by reading the messages here -- better tell the guy who said this because he gave the distinct impression otherwise. He seems confused, overall.

> Flatpak, DEB, PPA, __AppImage__, or a binary of any form, would be nice, yes please

**Comment by KaKi87** [comment]
> he gave the distinct impression otherwise

No. AppImage is one option among many others. I never said I included it because of being able to run on any Linux.

In fact, AppImage or another binary form are my least favourite options.

**Comment by genpfault** [comment]
> Thanks for your question. You are right, the README assumes that one is familiar with running python applications. You don't need a deb package or nix. You can clone this repo if you know how to do that with git or [download the source code](https://github.com/Elinvention/efiboots/archive/refs/heads/main.zip) and unzip it. Then double click efiboots (make it executable before if it isn't already).

There doesn't seem to be an `efiboots` file anywhere in the repo, as of (at least) [today's `main`](https://github.com/Elinvention/efiboots/tree/49812a396c7a567b9477c8cb338a3e8db317848a):

```none
$ git checkout 49812a396c7a567b9477c8cb338a3e8db317848a
$ git ls-files | grep efiboots
efiboots.spec
src/efiboots.gresource.xml
src/efiboots.in
test/test_efiboots.py
```

This procedure worked for me:

```none
# Build
meson setup --prefix=/tmp/efiboots builddir
meson compile -C builddir
meson install -C builddir

# Run
/tmp/efiboots/bin/efiboots
```
